### ISSUE TITLE
Unable to build RStudio with boost 1.74 due to problems related to bind updates in boost

### ISSUE THREAD

**Issue by ggardet** [rstudio/rstudio#7706]
Unable to build RStudio with boost 1.74 (was ok with 1.71). Due to 2 problems related to bind updates in boost:

1.  Missing `bind.hpp` include in `src/cpp/core/HtmlUtils.cpp` and `src/cpp/core/file_lock/FileLock.cpp`
2. And  the practice of declaring the Bind placeholders (_1, _2, ...) in the global namespace is deprecated. See boost message on compilation:
```
[  262s] /usr/include/boost/bind.hpp:36:1: note: '#pragma message: The practice of declaring the Bind placeholders (_1, _2, ...) in the global namespace is deprecated. Please use <boost/bind/bind.hpp> + using namespace boost::placeholders, or define BOOST_BIND_GLOBAL_PLACEHOLDERS to retain the current behavior.
```
Which finishes in:
```
[  275s] /home/abuild/rpmbuild/BUILD/rstudio-1.3.1073/src/cpp/core/system/PosixOutputCapture.cpp:83:79: error: '_1' was not declared in this scope
[  275s]    83 |       outHandler = boost::bind<void>(wrapHandler, stdoutHandler, dupStdoutFd, _1);
[  275s]       |                                                                               ^~
```

### System details

    RStudio Edition : 
    RStudio Version : 1.3.1073
    OS Version      :  openSUSE Tumbleweed
    R Version       : 

### Steps to reproduce the problem
Try to build RStudio with boost 1.74

### Describe the problem in detail
See above.

### Describe the behavior you expected
Build should be succesful.

**Comment by dcermak** [comment]
I think both are resolved via https://github.com/rstudio/rstudio/pull/7011, I will cherry pick that patch to RStudio in openSUSE.

**Comment by ggardet** [comment]
> I think both are resolved via #7011, I will cherry pick that patch to RStudio in openSUSE.

Indeed, but please note that defining `BOOST_BIND_GLOBAL_PLACEHOLDERS` is just a temp workaround and will stop to work at some point.
